compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -100,15 +95,22 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
                     }
                     FailureKind::MentionsParam => {
                         // FIXME(const_evaluatable_checked): Better error message.
-                        infcx
-                            .tcx
-                            .sess
-                            .struct_span_err(span, "unconstrained generic constant")
-                            .span_help(
+                        let mut err =
+                            infcx.tcx.sess.struct_span_err(span, "unconstrained generic constant");
+                        if let Ok(snippet) = infcx.tcx.sess.source_map().span_to_snippet(span) {
+                            err.span_suggestion(
+                                tcx.def_span(def.did),
+                                "try adding a `where` bound for this expression",
+                                format!("where {}: Sized", snippet),
```

Comment:
we might want to recommend this instead as it is otherwise always incorrect :thinking:
```suggestion
                                format!("where [u8; {}]: Sized", snippet),
```

using a suggestion seems a bit weird to me as you are pointing at the wrong `span`  here. I don't know how to get the span for the where clause of the current function though, so imo this is already an improvement.

Maybe also use a `span_help` here?

Can you add
```rust
// FIXME(const_evaluatable_checked): Update this suggestion once explicit const evaluatable bounds
// are implemented.

Reply:
I think adding a `span_help` after the suggestion would lead to very large and annoying errors, so I'll just leave 1 for now

Reply:
my idea was to use a help *instead* of a suggestion. The suggestion currently does not use the right span for the where clause so it will always be incorrect.